### PR TITLE
Support `.wat` files in the `explore` command

### PR DESCRIPTION
**Pull request by adambratschikaye**
Many on the `wasmtime` subcommands work with either `.wasm` or `.wat` files, but `explore` only works with `.wasm`. This change accepts `.wat` as well.

### CODE REVIEW

File: src/commands/explore.rs [hunk 1]
```diff
@@ -32,8 +32,10 @@ impl ExploreCommand {
 
         let config = self.common.config(self.target.as_deref())?;
 
-        let wasm = std::fs::read(&self.module)
+        let bytes = std::fs::read(&self.module)
             .with_context(|| format!("failed to read Wasm module: {}", self.module.display()))?;
+        let wasm = wat::parse_bytes(&bytes)
+            .with_context(|| format!("failed to parse Wasm module: {}", self.module.display()))?;
```
Comment (fitzgen) [inline src/commands/explore.rs:38]: Can you make sure that this also compiles and works for wasm binaries when the `wat` dependency isn't enabled? Something like the following:

```rust
                #[cfg(feature = "wat")]
                let bytes = wat::parse_bytes(bytes).map_err(|mut e| {
                    e.set_path(path);
                    e
                })?;
```

**Review by fitzgen — COMMENTED** [review]
Thanks! One nitpick below before we merge this.

File: src/commands/explore.rs (hunk 1 above)
Comment (adambratschikaye) [reply src/commands/explore.rs]: Sorry I forgot `wat` was behind a feature. Fixed it now.

**Review by fitzgen — APPROVED** [review]
Thanks very much!

**Pull request merged**